datalad/interface/ls.py

```diff
@@ -37,6 +37,8 @@
 from ..support.constraints import EnsureStr, EnsureNone
 from ..distribution.dataset import Dataset
 
+from datalad.distribution.subdatasets import Subdatasets
```

Comment:
This approach is errorprone since I or someone else might remove those imports since `Add` is used nowhere in this code and it requires really deep knowledge to know that such imports are really needed.  In some spots you did leave a comment to preclude removal,  but I am just saying that it is very suboptimal